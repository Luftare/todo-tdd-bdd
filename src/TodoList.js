import React from 'react';

export default ({
  items,
  addItem,
  newItemName,
  setNewItemName,
  deleteItem,
}) => (
  <div>
    <h1 data-title-e2e-test>Todo App</h1>

    <input
      value={newItemName}
      onChange={e => setNewItemName(e.target.value)}
      data-new-item-name-unit-test
      data-new-item-text-field-e2e-test
    />
    <button
      onClick={addItem}
      data-add-item-unit-test
      data-create-new-item-button-e2e-test
    >
      Add
    </button>
    <ul data-todo-list-e2e-test>
      {items.map(item => (
        <li key={item} data-todo-item-e2e-test={item}>
          <span>{item}</span>
          <button
            onClick={() => deleteItem(item)}
            data-delete-item-unit-test={item}
            data-delete-button-e2e-test={item}
          >
            x
          </button>
        </li>
      ))}
    </ul>
  </div>
);
