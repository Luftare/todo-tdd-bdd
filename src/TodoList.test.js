import React from 'react';
import { mount } from 'enzyme';

import TodoList from './TodoList';

describe('TodoList', () => {
  let component;
  let addItemStub;
  let setNewItemNameStub;
  let deleteItemStub;

  beforeEach(() => {
    addItemStub = () => {};
    setNewItemNameStub = jest.fn();
    deleteItemStub = jest.fn();

    component = mount(
      <TodoList
        items={['some-text-a', 'some-text-b', 'some-text-c']}
        setNewItemName={setNewItemNameStub}
        addItem={addItemStub}
        deleteItem={deleteItemStub}
      />
    );
  });

  it('renders correctly', () => {
    expect(component).toMatchHtmlSnapshot();
  });

  it('button for add item within, knows how to add item', () => {
    const button = component.find('button[data-add-item-unit-test]');

    expect(button).toHaveProp('onClick', addItemStub);
  });

  it('input for new item name within, knows how to set new item name', () => {
    const input = component.find('input[data-new-item-name-unit-test]');

    input.simulate('focus');
    input.simulate('change', { target: { value: 'something' } });

    expect(setNewItemNameStub).toHaveBeenCalledWith('something');
  });

  it('button to delete item within, knows how to delete item', () => {
    const deleteItemButton = component.find(
      'button[data-delete-item-unit-test="some-text-a"]'
    );

    deleteItemButton.simulate('click');

    expect(deleteItemStub).toHaveBeenCalledWith('some-text-a');
  });
});
