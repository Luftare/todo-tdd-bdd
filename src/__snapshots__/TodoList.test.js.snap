// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders correctly 1`] = `
<div>
  <h1
    data-title-e2e-test="true"
  >
    Todo App
  </h1>
  <input
    data-new-item-name-unit-test="true"
    data-new-item-text-field-e2e-test="true"
    value="some text"
  />
  <button
    data-add-item-unit-test="true"
    data-create-new-item-button-e2e-test="true"
  >
    Add
  </button>
  <ul
    data-todo-list-e2e-test="true"
  >
    <li
      data-todo-item-e2e-test="some-text-a"
    >
      <span>
        some-text-a
      </span>
      <button
        data-delete-button-e2e-test="some-text-a"
        data-delete-item-unit-test="some-text-a"
      >
        x
      </button>
    </li>
    <li
      data-todo-item-e2e-test="some-text-b"
    >
      <span>
        some-text-b
      </span>
      <button
        data-delete-button-e2e-test="some-text-b"
        data-delete-item-unit-test="some-text-b"
      >
        x
      </button>
    </li>
    <li
      data-todo-item-e2e-test="some-text-c"
    >
      <span>
        some-text-c
      </span>
      <button
        data-delete-button-e2e-test="some-text-c"
        data-delete-item-unit-test="some-text-c"
      >
        x
      </button>
    </li>
  </ul>
</div>
`;
